packages/snaps-sdk/src/jsx/components/Tooltip.ts

```diff
@@ -0,0 +1,46 @@
+import { createSnapComponent } from '../component';
+import type { StandardFormattingElement } from './formatting';
+import type { ImageElement } from './Image';
+import type { LinkElement } from './Link';
+import type { TextElement } from './Text';
+
+export type TooltipChildren =
+  | TextElement
+  | StandardFormattingElement
+  | LinkElement
+  | ImageElement
+  | null;
+
+/**
+ * The props of the {@link Tooltip} component.
+ *
+ * @property children - The children of the box.
+ * @property content - The text to display in the tooltip.
+ */
+export type TooltipProps = {
+  children: TooltipChildren;
+  content: TextElement | StandardFormattingElement | LinkElement | string;
+};
+
+const TYPE = 'Tooltip';
+
+/**
+ * A tooltip component, which is used to display text in a tooltip.
+ *
+ * @param props - The props of the component.
+ * @param props.children - The children of the tooltip.
+ * @param props.content - The text to display in the tooltip.
+ * @returns A tooltip element.
+ * @example
+ * <Tooltip content={<Text>foo</Text>}>
+ *   <Text>Hello world!</Text>
+ * </Tooltip>
```

Comment:
To make it clear that it takes strings too:
```suggestion
 * @example
 * <Tooltip content="Tooltip text">
 *   <Text>Hello world!</Text>
 * </Tooltip>
 * @example
 * <Tooltip content={<Text>Text with <Bold>formatting</Bold></Text>}>
 *   <Text>Hello world!</Text>
 * </Tooltip>
```